Test_Automation_Framework/Epam_TestAutomation_BusinessLogic/PageObjects/Pages/JoinOurTeamPages.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Epam_TestAutomation_Core.BasePage;
+using Epam_TestAutomation_Core.Browser;
+using Epam_TestAutomation_Core.Elements;
+using Epam_TestAutomation_Core.Helper;
+using Epam_TestAutomation_Core.Utils;
+using Epam_TestAutomation_TestData.JoinOurTeamInfo;
+using OpenQA.Selenium;
+
+namespace Epam_TestAutomation_BusinessLogic.PageObjects.Pages
+{
+    public class JoinOurTeamPages : BasePage
+    {
+        private SearchResultPages _searchResult;
+
+        public override bool IsOpened() => BrowserFactory.Browser.GetUrl().Equals(TestSettings.ApplicationUrl);
+
+        public Button CareersBlog => new Button(By.XPath("//*[@href='/careers/blog']"));
+
+        public Link CareerButton => new Link(By.XPath("//*[@href = '/careers']"));
+
+        public Button JobListingsButton => new Button(By.XPath("//*[@href = '/careers/job-listings']"));
+
+        public TextInput KeywordInput => new(By.XPath("//*[@id='new_form_job_search_1445745853_copy-keyword']"));
+
+        public Button FindButton => new(By.XPath("//*[@type='submit']"));
+
+        public Dropdown LocationDropdown => new Dropdown(By.XPath("//*[@class='select2-selection select2-selection--single']"));
+
+        public TextInput LocationInput => new TextInput(By.XPath("//*[@class='select2-search__field']"));
+
+        public Button CitiesLineButton => new Button(By.XPath("//*[@class='select2-results__option select2-results__option--highlighted']"));
+
+        public Label SkillsLabel => new Label(By.XPath("//*[@class='default-label']"));
+
+        public Label SkillFilter => new Label(By.XPath("//*[@class='filter-tag']"));
+
+        public Checkbox SkillsCheckBox(string skill) => new Checkbox(By.XPath($"//*[@class='checkbox-custom-label'][contains(text(), '{skill}')]"));
+
+        public Label ErrorMessage => new Label(By.XPath("//*[@class='search-result__error-message' and @role ='alert']"));
+
+        public JoinOurTeamPages()
+        {
+            _searchResult = new SearchResultPages();
+        }
+
+        public JoinOurTeamPages JoinOurTeamPagesIsOpened()
+        {
+            CareerButton.MoveToElement();
+            Waiters.WaitForCondition(CareersBlog.IsDisplayed);
+            JobListingsButton.Click();
+
+            return new JoinOurTeamPages();
+        }
+
+        public JoinOurTeamPages GetProfessionKeyword(string keyword)
+        {
+            KeywordInput.SendKeys(keyword);
+            FindButton.Click();
+            Waiters.WaitForCondition(_searchResult.SearchResultTitle.IsDisplayed);
+
+            return new JoinOurTeamPages();
+        }
+
+        public JoinOurTeamPages GetLocationKeyword(string keyword)
+        {
+            LocationDropdown.Click();
+            LocationInput.SendKeys(keyword);
+            CitiesLineButton.Click();
+
+            return new JoinOurTeamPages();
+        }
+
+        public JoinOurTeamPages GetSkillKeyword(string keyword)
```

Comment:
You can create one method for all search parameters and and verification 
if (profession != null)
{
KeywordInput.SendKeys(profession);
}
the same for another parameters. And delete all another methods.

Reply:
Any updates?